dags/als-kp-dataset-dag.py

```diff
@@ -0,0 +1,338 @@
+                item["description"][:1000]
+                if len(item["description"]) > 1000
+                else item["description"]
+            )
+            dataset = Dataset(
+                name=item["title"],
+                description=dataset_description,
+                parent_id=context["params"]["project_id"],
+            ).store(synapse_client=synapse_client)
+            dataset_collection.add_item(dataset)
+
+        dataset_collection = dataset_collection.store(synapse_client=synapse_client)
+        return dataset_collection
+
+    @task
+    def update_annotations(
+        dataset_collection: DatasetCollection,
+        transformed_items: List[Dict[str, Any]],
+        **context,
+    ) -> None:
+        """Update dataset annotations and create snapshot if changes are detected.
+
+        This task:
+        1. Queries the current state of the dataset collection
+        2. Prepares new annotation data from the transformed items
+        3. Updates the collection with the new annotations
+        4. Queries the updated state
+        5. Compares the before and after states
+        6. If changes are detected:
+           - Creates a summary of which columns were modified
+           - Creates a new snapshot with a descriptive comment
+           - The comment includes timestamp, changed columns, and dataset count
+
+        Args:
+            dataset_collection: The dataset collection to update
+            transformed_items: List of transformed items containing new annotations
+            **context: Airflow task context containing DAG parameters
+        """
+        syn_hook = SynapseHook(context["params"]["synapse_conn_id"])
+        synapse_client = syn_hook.client
+
+        dataset_ids = [item.id for item in dataset_collection.items]
+
+        # Get current data before update
+        current_data = dataset_collection.query(
+            query=f"SELECT * from {dataset_collection.id}",
+            synapse_client=synapse_client,
+        )
+        current_df = pd.DataFrame(current_data)
+
+        # Prepare and apply new data
+        annotation_data = pd.DataFrame(
+            {
+                "id": dataset_ids,
+                **{
+                    key: [
+                        (
+                            ", ".join(item[key])
+                            if isinstance(item[key], list)
+                            else item[key]
+                        )
+                        for item in transformed_items
+                    ]
+                    for key in [
+                        "title",
+                        "creator",
+                        "keywords",
+                        "subject",
+                        "collection",
+                        "publisher",
+                        "species",
+                        "sameAs",
+                    ]
+                },
+            }
+        )
+
+        # Update the rows
+        dataset_collection.update_rows(
```

Comment:
I think you'll want to set this to true

https://github.com/Sage-Bionetworks/synapsePythonClient/blob/70e53e692673930a7c3ce7882aec5faa94632bad/synapseclient/models/dataset.py#L228